### PR TITLE
Adds test to try and break the move API

### LINKED ISSUE #79: Ingest API /move tasks fail with multiple files.
While using the <ingest server>/move api we found that with multiple files in the json data there were times when we checked the status of the job we would get an error such as this:
```
Traceback (most recent call last):
  File "/opt/indms/lib/python2.7/site-packages/pacifica/ingest/tasks.py", line 75, in move_files
    patch_files(meta_obj)
  File "/opt/indms/lib/python2.7/site-packages/pacifica/ingest/tarutils.py", line 287, in patch_files
    data=json.dumps(data)
  File "/opt/indms/lib/python2.7/site-packages/requests/sessions.py", line 605, in patch
    return self.request('PATCH', url, data=data, **kwargs)
  File "/opt/indms/lib/python2.7/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/opt/indms/lib/python2.7/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/opt/indms/lib/python2.7/site-packages/requests/adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
ConnectionError: ('Connection aborted.', BadStatusLine("''",))
```

We eventually took some tcpdumps:

[failure.pcap.gz](https://github.com/pacifica/pacifica-ingest/files/2933475/failure.pcap.gz)
[worky.pcap.gz](https://github.com/pacifica/pacifica-ingest/files/2933476/worky.pcap.gz)

You should be able to see in there that the failure pcap file seems to try to re-use the connection on the /PATCH call that fails.

In order to work around the re-use of connections we changed the patch_files code here:
https://github.com/pacifica/pacifica-ingest/blob/8bcaac01f0f1655739c0fde6d159311b01814b44/pacifica/ingest/tarutils.py#L276-L291

on line 285 to be:
```
 req = requests.patch( 
```
which makes the multi-file calls work reliably, but throws away any re-try logic in a session.

please find an appropriate fix

### PR DESCRIPTION
**Pull request by dmlb2000**
### Description

This is the first step in fixing #79 hopefully this test can break
the move API by forcing the ingest backend to try and and reuse a
open socket connection to the archive interface.

Signed-off-by: David Brown <dmlb2000@gmail.com>

### Issues Resolved

Fix #79  

### Check List

- [ ] All tests pass. See <https://github.com/pacifica/pacifica-docs/blob/master/development/TESTING.md>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -70,7 +72,7 @@ env:
 - pacifica-metadata-cmd dbsync
 - pacifica-uniqueid-cmd dbsync
 - pacifica-metadata & echo $! > metadata.pid
-- pacifica-archiveinterface & echo $! > archivei.pid
+- uwsgi --http-socket :9080 --master --module 'pacifica.archiveinterface.wsgi' & echo $! > archivei.pid
```
Comment (dmlb2000) [inline .travis.yml:75]: @karcaw I've converted the archiveinterface to uwsgi.

File: tests/move_test.py [hunk 2]
```diff
@@ -13,3 +24,51 @@ def test_good_move():
...
+            'mtime': 'Tue Nov 29 14:09:05 PST 2016',
+            'name': 'tempfile.{}.txt'.format(file_index),
+            'size': os.stat(file_name).st_size,
+            'source': file_name,
+            'subdir': 'data/a/b/'
+        })
+    yield file_list
+
+
+def test_big_move():
```
Comment (dmlb2000) [inline tests/move_test.py:58]: @karcaw Here's where I moved 100 temporary files into pacifica.

File: .travis.yml [hunk 3]
```diff
@@ -174,3 +177,5 @@ jobs:
         tags: true
 install:
 - pip install -r requirements-dev.txt
+- pip install uwsgi
+- pip freeze
```
Comment (dmlb2000) [inline .travis.yml:181]: @karcaw can you verify the versions you are using?

**Review by dmlb2000 — COMMENTED** [review]
@karcaw I've converted the archive interface to using uwsgi and attempted a move upload of 100 files. However, travis can only test to localhost.

File: .travis.yml (hunk 1 above)
Comment (karcaw) [reply .travis.yml:75]: nice

File: tests/move_test.py (hunk 2 above)
Comment (karcaw) [reply tests/move_test.py:58]: nice

File: .travis.yml (hunk 3 above)
Comment (karcaw) [reply .travis.yml:181]: versions of what?

Comment (dmlb2000) [reply .travis.yml:181]: This is the `pip freeze` output from Travis

```
alabaster==0.7.12
amqp==2.4.2
aspy.yaml==1.2.0
astroid==1.6.6
atomicwrites==1.2.1
attrs==18.2.0
Babel==2.6.0
backports.functools-lru-cache==1.5
backports.tempfile==1.0
backports.weakref==1.0.post1
billiard==3.6.0.0
celery==4.3.0
certifi==2018.10.15
cfgv==1.6.0
chardet==3.0.4
cheroot==6.5.5
CherryPy==18.1.1
colorama==0.4.1
commonmark==0.8.1
coverage==4.5.3
docutils==0.14
elasticsearch==7.0.0
entrypoints==0.3
flake8==3.7.7
flake8-polyfill==1.0.2
future==0.17.1
identify==1.4.2
idna==2.8
imagesize==1.1.0
importlib-metadata==0.9
importlib-resources==1.0.2
isort==4.3.17
jaraco.functools==2.0
Jinja2==2.10.1
kombu==4.5.0
lazy-object-proxy==1.3.1
mando==0.6.4
MarkupSafe==1.1.1
mccabe==0.6.1
mock==2.0.0
more-itertools==4.3.0
nodeenv==1.3.3
nose==1.3.7
numpy==1.16.2
pacifica-archiveinterface==0.2.0
pacifica-metadata==0.9.2
pacifica-policy==0.5.2
pacifica-uniqueid==0.3.3
packaging==19.0
pbr==5.0.0
peewee==3.9.5
pep257==0.7.0
pipenv==2018.11.26
pluggy==0.8.0
portend==2.4
pre-commit==1.15.2
psycopg2==2.7.7
py==1.7.0
pycodestyle==2.5.0
pyflakes==2.1.1
Pygments==2.3.1
pylint==1.9.4
PyMySQL==0.9.3
pyparsing==2.4.0
pytest==4.3.1
python-dateutil==2.8.0
pytz==2019.1
PyYAML==5.1
radon==3.0.1
recommonmark==0.5.0
requests==2.21.0
six==1.12.0
snowballstemmer==1.2.1
Sphinx==2.0.1
sphinxcontrib-applehelp==1.0.1
sphinxcontrib-devhelp==1.0.1
sphinxcontrib-htmlhelp==1.0.2
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.2
sphinxcontrib-serializinghtml==1.1.3
tempora==1.14.1
toml==0.10.0
urllib3==1.24.2
uWSGI==2.0.18
vine==1.3.0
virtualenv==16.0.0
virtualenv-clone==0.4.0
wrapt==1.11.1
zc.lockfile==1.4
zipp==0.4.0
```

The `cherrypy` and `uWSGI` packages are key here.

**Comment by dmlb2000** [comment]
@karcaw the take away here was that I was unable to reproduce your issue. However, we do have a better stress test for the move API. This can also be applied to the ingest API at some point as well.

**Pull request merged**